src/mysql.erl

```diff
@@ -198,6 +196,10 @@ backported_gen_server_stop(Conn, Reason, Timeout) ->
         end
     end.
 
+-spec is_connected(Conn) -> boolean()
```

Comment:
One line of edoc wouldn't hurt, e.g.

```Erlang
%% @doc Always returns true, except for connections started with
%% connect mode `lazy' or `asynchronous' where false can be retuned
%% before a connection has been established.
```

Reply:
This is not entirely accurate. Even in asynchronous mode, it will always return `true`: The connection process spawns the temporary process that does the actual connect to the server and waits for completion in a `receive`. Thus, it won't react to any message sent to it (like with `gen_server:call`) while it waits at this point, and when it is finally able to react, it also means the connection has been established.

Now that I think about it, what _may_ happen is that `is_connected/1` times out, depending on how long the connect takes in asynchronous mode, or if the connection process is busy executing a query or something in all modes. So I better provide a `_is_connected/2` which takes a timeout, yes?

Reply:
Nah, not worth it, is it? Just put a `%% @private` so it doesn't show up in the documentation.

Reply:
Ok, done in latest commit.

Reply:
Aha :) i meant @private to hide the function entirely, so we don't need to document it. Not great if it's i correct i guess. Not a big deal though.

Reply:
Aha :) i meant @private to hide the function entirely, so we don't need to document it. Not great if it's i correct i guess. Not a big deal though.

Reply:
Aha :) i meant @private to hide the function entirely, so we don't need to document it. Not great if it's i correct i guess. Not a big deal though.

Reply:
Ah, ok, I'll remove the documentation again then.